### PR TITLE
refactor kindnetd

### PR DESCRIPTION
**Pull request by BenTheElder**
this breaks things up a little bit and does some misc cleanup

future steps:
- refactor to actually be a proper controller (not super important right now, but a better plan long term...)
- tag routes with kindnetd as the source so we can cleanup after deleted nodes (low priority, but we should consider this future direction for robustness when developing it...)

### CODE REVIEW

File: pkg/build/node/cni.go [hunk 1]
```diff
@@ -185,21 +186,23 @@ spec:
...
-          - name: POD_IP
-            valueFrom:
-              fieldRef:
-                fieldPath: status.podIP
-          - name: CNI_CONFIG_TEMPLATE
-            valueFrom:
-              configMapKeyRef:
-                name: kindnet-cfg
-                key: cni-conf-template.json
+        - name: HOST_IP
```
Comment (aojea) [inline pkg/build/node/cni.go]: is this well indented? seem indentation has changed

Comment (BenTheElder) [reply pkg/build/node/cni.go]: this normalized the indentation, see EG the volumeMounts below

**Comment by BenTheElder** [comment]
next up going to refactor the build, iterating on building the image is very slow right now, but should be easy to fix 😅

File: cmd/kindnetd/main.go [hunk 2]
```diff
@@ -94,29 +62,28 @@ func main() {
...
+	// enforce ip masquerade rules
 	ipv6 := false
+	// TODO: dual stack...?
 	if strings.Contains(hostIP, ":") {
 		ipv6 = true
 	}
-	// used to create routes and iptables rules
-	kindnetd := NewKindnet(cniConfigWriter, ipv6, kindnetConfig)
+	masqAgent, _ := NewIPMasqAgent(ipv6, []string{os.Getenv("POD_SUBNET")})
+	go masqAgent.SyncRulesForever(time.Second * 60)
```
Comment (aojea) [inline cmd/kindnetd/main.go]: what happens if there is an error with the masqAgent?

Comment (BenTheElder) [reply cmd/kindnetd/main.go]: right now it won't handle it. we could panic or log an error.

I think in this case we should panic just until a (near) future refactor adds proper logging

Comment (BenTheElder) [reply cmd/kindnetd/main.go]: updated to panic for now with a TODO to log pervasively.

xref: #109 I think it's about time to consider merging with the ecosystem on that front...

Comment (BenTheElder) [reply cmd/kindnetd/main.go]: thanks for catching this.

File: cmd/kindnetd/masq.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	for {
+		if err := ma.SyncRules(); err != nil {
+			return err
+		}
+		time.Sleep(interval)
+	}
+}
+
+// name of nat chain for iptables masquerade rules
+const masqChainName = "KIND-MASQ-AGENT"
```
Comment (BenTheElder) [inline cmd/kindnetd/masq.go]: this is now fixed instead of a flag (the flag came from the original masq agent)

File: cmd/kindnetd/masq.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		noMasqueradeCIDRs: noMasqueradeCIDRs,
+	}, nil
+}
+
+// IPMasqAgent is based on https://github.com/kubernetes-incubator/ip-masq-agent
+// but collapsed into kindnetd and made ipv6 aware in an opionated and simplified
+// fashion
+type IPMasqAgent struct {
+	iptables          utiliptables.Interface
+	masqChain         utiliptables.Chain
```
Comment (BenTheElder) [inline cmd/kindnetd/masq.go]: this chain is now in the agent instead of a global

**Comment by aojea** [comment]
outstanding 🥇

**Pull request merged**